samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt

```diff
@@ -17,101 +17,105 @@ package com.google.oboe.sample.drumthumper
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.widget.Button
 
-class DrumThumperActivity : AppCompatActivity(), DrumPad.DrumPadTriggerListener {
+class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListener {
+    private val TAG = "DrumThumperActivity"
+
     private var mDrumPlayer = DrumPlayer()
 
     init {
+        // Load the library containing the a native code inxlusinf rhw JNI  functions
         System.loadLibrary("drumthumper")
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-
-        mDrumPlayer.init();
+        mDrumPlayer.setupAudioStream();
```

Comment:
You should probably open and close the audio streams in onStart() or onResume()
and close in onPause() or onStop()
so that you do not hold the EXCLUSIVE MMAP resource.